### PR TITLE
feat(api_models): add client_secret type to payments

### LINKED ISSUE #1357: [FEATURE] add domain type for client secret
### Feature Description

Create domain type for client secret

### Possible Implementation

Make type `ClietSecret(String)` and use it wherever applicable

### Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### Are you willing to submit a PR?

No, but I'm happy to collaborate on a PR with someone else

### PR DESCRIPTION
**Pull request by wowinter13**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Implement `ClientSecret` type for payments. 

Closes #1357 

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
Added and ran tests successfully

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [x] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by wowinter13** [comment]
@SanchithHegde @deepanshu-iiitu @Abhicodes-crypto hello there guys
Could you check please if I am going in the right direction?

File: crates/api_models/src/payments.rs [hunk 1]
```diff
@@ -51,6 +54,113 @@ pub struct BankCodeResponse {
...
+    where
+        D: Deserializer<'de>,
+    {
+        struct ClientSecretVisitor;
+
+        impl<'de> Visitor<'de> for ClientSecretVisitor {
+            type Value = ClientSecret;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+                formatter.write_str("a string in the format 'payment_id_secret_secret'")
```
Comment (Narayanbhat166) [inline crates/api_models/src/payments.rs:74]:
```suggestion
                formatter.write_str("a string in the format '{payment_id}_secret_{secret}'")
```

**Comment by Narayanbhat166** [comment]
apart from the minor nit, LGTM!

File: crates/api_models/src/payments.rs (hunk 1 above)
Comment (wowinter13) [reply crates/api_models/src/payments.rs]: @Narayanbhat166 fixed, thanks!

**Review by Narayanbhat166 — DISMISSED** [review]

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @wowinter13!

**Comment by SanchithHegde** [comment]
Hey @wowinter13, could you address the failing CI checks?

**Review by SanchithHegde — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Pull request merged**